doc/sonic-application-extension/tpcm_app_ext.md

```diff
@@ -0,0 +1,467 @@
+        "name": "default"
+    },
+    "service": {
+        "name": "default",
+        "requires": [
+            "docker"
+        ],
+        "after": [
+            "docker"
+        ],
+        "before": [],
+        "dependent-of": [],
+        "asic-service": false,
+        "host-service": false,
+        "warm-shutdown": {
+            "after": [],
+            "before": []
+        },
+        "fast-shutdown": {
+            "after": [],
+            "before": []
+        },
+        "syslog": {
+            "support-rate-limit": true
+        }
+    },
+    "container": {
+        "privileged": false,
+        "volumes": [],
+        "tmpfs": [],
+        "memory": "512M",
+        "cpu-quota": "20000",
+        "cpu-period": "100000",
+        "command": ""
+    },
+    "cli": {
+        "config": "",
+        "show": "",
+        "clear": ""
+    },
+    "tpcm": true
+}
+
+```
+- The default manifest file contains specifications allowing TPC packages to utilize 20% of the CPU and up to 512MB of memory.
+
+
+
+#### TPC Update
+
+- The manifest file properties that can be updated for a TPCM Docker package are:
+```
+'privileged', 'memory', 'cpu-quota', 'cpu-period', 'command' under container category
+'requires', 'after', 'before', 'asic-service', 'host-service' under service category.
+```
+
+- The local manifest file can be updated using "sonic-package-manager manifests update" command
+
+- The 'command' attribute is used to update the docker startup command arguments. For example, setting "command":"--path.rootfs=/host" would configure the host filesystem as the root filesystem path for the container
+
+- After updating the local manifest file, the TPC container can be updated with the "sonic-package-manager update" command. This command will reinstall the TPC container using the new manifest file.
+
+
+#### TPC Uninstall
+
+- The uninstallation process for TPC packages using 'sonic-package-manager uninstall' follows the same procedure as regular Sonic packages, 
+    where the corresponding Docker container and service are stopped and removed, and the package image is deleted if it is not in use. 
+    The local manifest file associated with the TPC package is also removed during uninstallation.
+
+
+#### TPC Upgrade
+
+- The upgrade process for TPC packages using 'sonic-package-manager install' follows the same procedure as regular Sonic packages. 
+- Since the manifest package version mismatch is the trigger for package upgrade, the user must use --force option for TPC upgrade.
+- The new version of the TPC package is downloaded and installed, and the new manifest is created if necessary. 
+- The old version of the TPC package is then removed if not in use by other package, and the running container is restarted with the new version.
+
+
+#### Resource Limits  for TPC
+
+Following system resource limits shall be applicable to TPCs:
+- CPU 
+- System Memory
+
+These limits can be configured during the TPC installation. 
```

Comment:
I think, resource limits are good options which need not be restricted only for TPC.

Reply:
@sg893052 @venkatmahalingam Agree, shouldn't be restricted for TPC. In fact, resource limits were discussed to be added to regular app.ext when it first was introduced but this functionality was delayed since we had no use for configuring limits.

Reply:
The inclusion of resource limits for TPC within the CONFIG DB is a distinct feature that can also be extended to SONiC containers. Therefore, the resource limits aspect will be addressed in a subsequent phase.